libraries/lib-track/Track.cpp

```diff
@@ -721,6 +721,14 @@ bool TrackList::MakeMultiChannelTrack(Track& track, int nChannels, bool aligned)
          return false;
 
       (*first)->SetLinkType(aligned ? Track::LinkType::Aligned : Track::LinkType::Group);
+
+      //Cleanup the group data in all channels except the first
+      for(auto it = std::next(first), last = std::next(first, nChannels);
+         it != last;
+         ++it)
+      {
+         (*it)->DestroyGroupData();
+      }
```

Comment:
Your commit comment suggest that you think this is redundant. I think you are right after all.  I overlooked other details.  But the extra insurance is harmless.